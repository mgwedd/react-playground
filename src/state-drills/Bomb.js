import React, { Component } from 'react';

export default class Bomb extends Component {
    
    state = {
        bomb: '',
        interval: 0
    }

    componentDidMount() {
        this.interval = setInterval(() => {
            this.setState({interval: this.state.interval + 1})
            const divBy2 = this.state.interval % 2 === 0;
            const isOrAbove8 = this.state.interval >= 8;
            if (isOrAbove8) {
                this.setState({bomb: 'BOOOOOM !!!'})
            } else if (divBy2) {
                this.setState({bomb: 'tick'})
            } else {
                this.setState({bomb: 'tock'})
            }
        }, 1000);
    }

    componentWillUnmount() {
        clearInterval(this.interval);
    }
    render() {
        return (
            <div>
                <p>{this.state.bomb}</p>
            </div>
        );
    }
}