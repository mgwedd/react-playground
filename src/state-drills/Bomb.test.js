import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Bomb from './Bomb';

describe('Bomb Component', () => {

    it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Bomb />, div);
    ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
    const tree = renderer
        .create(<Bomb />)
        .toJSON();
    expect(tree).toMatchSnapshot();  
    });
    
});