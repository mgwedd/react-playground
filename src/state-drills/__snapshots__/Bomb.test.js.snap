// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bomb Component renders the UI as expected 1`] = `
<div>
  <p>
    
  </p>
</div>
`;
